blindml/frontend/config/task/task.py

```diff
@@ -75,6 +77,8 @@ def __init__(self, task_fp) -> None:
                     - set(self._task.payload.drop_cols)
                     - {self._task.payload.y_col}
                 )
+                # ^ should this drop the y_col even when there isn't a drop_cols
+                # configuration?
```

Comment:
yes because otherwise we're training on the `y_col` to predict `y_col` (i.e. poisoning the training)

Reply:
i guess if the user specifies the ycol in their explicitly specified X_cols, that's their own fault, though?

Reply:
yup major party foul (but there should also probably be a check)

Reply:
it looks like the explicit X_cols code path was broken anyway - I've put in a fix and added a test case.

I've also added in an assert in the explicit X_cols path for "no ycol in xcols"